libs/remixd/src/services/gitClient.ts

```diff
@@ -0,0 +1,62 @@
+import * as WS from 'ws' // eslint-disable-line
+import { PluginClient } from '@remixproject/plugin'
+const { spawn } = require('child_process')
+const gitRegex = '^git\\s[^&|;]*$'
+
+export class GitClient extends PluginClient {
+  methods: ['command']
+  websocket: WS
+  currentSharedFolder: string
+  readOnly: boolean
+
+  setWebSocket (websocket: WS): void {
+    this.websocket = websocket
+  }
+
+  sharedFolder (currentSharedFolder: string, readOnly: boolean): void {
+    this.currentSharedFolder = currentSharedFolder
+    this.readOnly = readOnly
+  }
+
+  command (cmd: string) {
+    return new Promise((resolve, reject) => {
```

Comment:
I would recommend to put the Promise closer to its used : 
```typescript
assertCommand(cmd, gitRegex)
const options = { cwd: this.currentSharedFolder, shell: true }
const child = spawn(cmd, options)
let result = ''
let error = ''
return new Promise((resolve, reject) => {
  child.stdout.on('data', (data) => result += data.toString())
  child.stderr.on('data', (err) => error += data.toString())
  child.on('close', (exitCode) => exitCode !== 0 ? reject(error) : resolve(result))
})
```
You don't need to try / catch here everything that throw will be handled by the caller